core/src/main/scala/kafka/log/Log.scala

```diff
@@ -565,19 +565,24 @@ class Log(val dir: File,
   def deleteOldSegments(predicate: LogSegment => Boolean): Int = {
     // find any segments that match the user-supplied predicate UNLESS it is the final segment
     // and it is empty (since we would just end up re-creating it
-    val lastSegment = activeSegment
-    val deletable = logSegments.takeWhile(s => predicate(s) && (s.baseOffset != lastSegment.baseOffset || s.size > 0))
-    val numToDelete = deletable.size
-    if(numToDelete > 0) {
```

Comment:
activeSegment is used by other methods. Changing activeSegment to return null is not nice and will force us to add null checks to other methods.  I have added  (segments.lastEntry != null) check to deleteOldSegments() method. activeSegment/lastEntry will be null only after log delete() call.

Pl review latest update